code/modules/mod/modules/modules_general.dm

```diff
@@ -137,41 +142,21 @@
 /obj/item/mod/module/jetpack/configure_edit(key, value)
 	switch(key)
 		if("stabilizers")
-			stabilizers = text2num(value)
-
-/obj/item/mod/module/jetpack/proc/move_react(mob/user)
-	SIGNAL_HANDLER
+			set_stabilizers(text2num(value))
 
-	if(!active)//If jet dont work, it dont work
-		return
-	if(!isturf(mod.wearer.loc))//You can't use jet in nowhere or from mecha/closet
-		return
-	if(!(mod.wearer.movement_type & FLOATING) || mod.wearer.buckled)//You don't want use jet in gravity or while buckled.
-		return
-	if(mod.wearer.pulledby)//You don't must use jet if someone pull you
-		return
-	if(mod.wearer.throwing)//You don't must use jet if you thrown
-		return
-	if(user.client && length(user.client.keys_held & user.client.movement_keys))//You use jet when press keys. yes.
-		allow_thrust()
-
-/obj/item/mod/module/jetpack/proc/pre_move_react(mob/user)
-	SIGNAL_HANDLER
-
-	ion_trail.oldposition = get_turf(src)
-
-/obj/item/mod/module/jetpack/proc/spacemove_react(mob/user, movement_dir)
-	SIGNAL_HANDLER
-
-	if(active && (stabilizers || movement_dir))
-		return COMSIG_MOVABLE_STOP_SPACEMOVE
-
-/obj/item/mod/module/jetpack/proc/allow_thrust()
+/obj/item/mod/module/jetpack/proc/allow_thrust(use_fuel = TRUE)
+	if(!use_fuel)
+		return check_power(use_power_cost)
 	if(!drain_power(use_power_cost))
-		return
-	ion_trail.generate_effect()
+		return FALSE
 	return TRUE
 
+/obj/item/mod/module/jetpack/proc/get_user()
+	var/mob/user = mod?.wearer
+	if(!ismob(user))
+		return null
+	return user
```

Comment:
maybe i'm not understanding of modsuit sanity but there are multiple things here that confuse me:
1. `get_user` will be called when this is outside of a modsuit, making `mod?` necessary?
2. wearer will not be a mob? I mean, considering the above and the fact that the module will only work in an activated modsuit, that's essentially saying this will get called by a limbed mob at the very least
3. finally, considering the above two questions, does it matter to return null specifically if that check fails, considering it could only fail if mod.wearer is null?

Reply:
yeah. under the circumstances this is ran on, mod will always be present, and user is always a human mob.

Reply:
I'm trying to make this proc runtime safe, but maybe that's a bad idea. whatcah think?

Reply:
Removed the runtime safety, kept the typechecking since we want that to fail early